pub-critic/src/app/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,85 @@
+"use client";
+import React, { useEffect, useState } from "react";
+import classes from "./Dropdown.module.scss";
+interface Option {
+  label: string;
+  value: string | number;
+}
+
+interface Props {
+  options: Option[];
+  onSelect: (value: string | number) => void;
+  cancel?: boolean;
+  closer?: Function;
+}
+
+export const Dropdown = ({ options, onSelect, cancel, closer }: Props) => {
+  const [searchTerm, setSearchTerm] = useState<string | number>("");
+  const [selected, setSelected] = useState<string | number>("");
+  const [visible, setVisible] = useState<boolean>(false);
+  const [placeholder, setPlaceholder] = useState<string>("Search");
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value);
+  };
+
+  const handleSelect = (value: string | number, label: string) => {
+    setSelected(value);
+    setSearchTerm("");
+    setPlaceholder(label);
+    onSelect(value);
+    setVisible(false);
+  };
+
+  useEffect(() => {
+    setVisible(false);
+  }, [cancel]);
+
+  return (
+    <div className={classes.dropdown}>
+      <div
+        className={`${visible ? classes.active : ""} ${classes.top} `}
+        onClick={() => {
+          closer && closer();
+          setVisible((prev) => !prev);
+        }}
```

Comment:
`onClick={handleClick}`

ps always avoid arrow expressions in the Render method

Reply:
✅